src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -28,8 +29,9 @@ public partial class RegexGenerator
             Compilation compilation = sm.Compilation;
             INamedTypeSymbol? regexSymbol = compilation.GetBestTypeByMetadataName(RegexName);
             INamedTypeSymbol? generatedRegexAttributeSymbol = compilation.GetBestTypeByMetadataName(GeneratedRegexAttributeName);
+            INamedTypeSymbol? intSymbol = compilation.GetBestTypeByMetadataName(Int32Name);
```

Comment:
You probably want to use `.Type.SpecialType == SpecialType.System_Int32` instead.